### PR TITLE
Documentation updates 

### PR DESCRIPTION
**Pull request by danbadge**
Trying to make it clearer as to which implementation is mock and real whilst also updating documented version number when including in deps.

Let me know if I've missed anything with the above.

Any thing else we need to update here?

### CODE REVIEW

**Review by amencarini — APPROVED** [review]
LGTM!

**Comment by amencarini** [comment]
Actually, given you're diving on this one:

https://github.com/Driftrock/mockingbird/blob/master/lib/mockingbird/fake_client.ex

The file name does not match the module name and it has no `__using__` macro, so the `use Mockingbird.Client` that's on the readme won't work, needs to be `import` rather than `use`.

**Comment by danbadge** [comment]
Leaving this open for now as I want to have a look at explaining in the README why we do this kind of testing

**Comment by amencarini** [comment]
> The problem we encountered with [switching client modules depending on environment](http://blog.plataformatec.com.br/2015/10/mocks-and-explicit-contracts/), which is the prevalent solution in the community, is that we found our tests were not exercising the production client module enough; a few times we changed the test client module and miss the required changes in the production client module. Hence we decided to drop one level deeper and switch `HTTPoison` while testing instead.

File: README.md [hunk 1]
```diff
@@ -6,6 +6,12 @@
...
+
+The prevalent solution in the Elixir community for this type of testing is to [switch external client modules for fake implementations depending on the environment](http://blog.plataformatec.com.br/2015/10/mocks-and-explicit-contracts/). This solution is simple and works nicely, however we found that our tests were not exercising enough of the production external client to provide us with the safety we needed when making changes. It's too easy to fall into the trap of only making changes to the test client and not replicating those in the production client. So we decided to drop one level deeper and switch our HTTP Client (typically `HTTPPoison`) for a fake implementation when testing. Mockingbird is our solution for that.
+
```
Comment (danbadge) [inline README.md:14]: @amencarini what do you think?

Comment (amencarini) [reply README.md:14]: image

**Pull request merged**